assets/src/scripts/monitoring-location/components/hydrograph/selectors/legend-data.js

```diff
@@ -32,61 +34,81 @@ const getLegendDisplay = createSelector(
     isWaterwatchVisible,
     getWaterwatchFloodLevels,
     getUniqueGWKinds,
-    (showCompare, showMedian, medianSeries, currentClasses, compareClasses, showWaterWatch, floodLevels, gwLevelKinds) => {
+    getPrimaryParameter,
+    (showCompare, showMedian, medianSeries, currentClasses, compareClasses, showWaterWatch, floodLevels, gwLevelKinds,
```

Comment:
A particular kind of data has three states: Not requested (or available for the site) so the selector will return null; Data was requested but there is no values for the current hydrograph settings so the selector for that kind of data will return an empty array; and Data was requested and there are values, the selector will return a non-zero length array. That's how the legend determines whether to show the No data message.